### PR TITLE
HPOS compatibility

### LINKED ISSUE #420: HPOS will change screen ID for the orders page, we need to support that
The new HPOS feature scheduled for WC version 7.1 [will change the order page screen ID](https://developer.woocommerce.com/2022/10/11/hpos-upgrade-faqs/) from `shop_order` to `woocommerce_page_wc-orders` which means that this hooks must be updated keeping in mind the <7.1 versions as well:

`manage_edit-shop_order_columns` => `manage_woocommerce_page_wc-orders_columns`
`manage_shop_order_posts_custom_column` => `manage_woocommerce_page_wc-orders_custom_column`
`manage_edit-shop_order_sortable_columns` => `manage_woocommerce_page_wc-orders_sortable_columns`

Also check other places where the screen ID is checked and make sure it also complies with the new one, specially meta boxes display.

### PR DESCRIPTION
**Pull request by alexmigf**
closes #420

### How to test?

You can test HPOS now using WooCommerce version 7.1, no need to install a special version anymore. You should follow this steps: https://developer.woocommerce.com/2022/09/16/high-performance-order-storage-call-for-testing-vol-2/#download-and-install

HPOS Sync is still much under development, it might not work as expected, I also had issues with it which I posted here: https://github.com/woocommerce/woocommerce/issues/34981

Desired testing setups:

* Existing WC <=7.0 setup + WC 7.1 upgrade + HPOS **enabled** + Sync **enabled** (can have issues)
* New WC 7.1 setup + HPOS **enabled** + Sync **disabled**
* Existing WC <7.1 setup + HPOS **disabled**
* New WC 7.1 setup + HPOS **disabled**

By "existing" I mean older versions of WC prior to 7.1 with orders created in the `wp_posts` table.

### CODE REVIEW

File: includes/class-wcpdf-admin.php [hunk 1]
```diff
@@ -1,22 +1,34 @@
...
 
 class Admin {
+
 	function __construct()	{
 		add_action( 'woocommerce_admin_order_actions_end', array( $this, 'add_listing_actions' ) );
 
-		add_filter( 'manage_edit-shop_order_columns', array( $this, 'add_invoice_columns' ), 999 );
-		add_action( 'manage_shop_order_posts_custom_column', array( $this, 'invoice_columns_data' ), 2 );
-		add_filter( 'manage_edit-shop_order_sortable_columns', array( $this, 'invoice_columns_sortable' ) );
+		if ( defined( 'WC_VERSION' ) && version_compare( WC_VERSION, '7.1', '>=' ) ) {
```
Comment (DavidAnderson684) [inline includes/class-wcpdf-admin.php:18]: @alexmigf Why is the condition and branching needed? Presumably the actions/filters are only called when needed, so they don't need to be conditional.

File: includes/class-wcpdf-admin.php [hunk 2]
```diff
@@ -307,16 +313,18 @@ public function request_query_sort_by_column( $query_vars ) {
 	}
 
 	/**
-	 * Add the meta box on the single order page
+	 * Add the meta boxes on the single order page
 	 */
 	public function add_meta_boxes() {
+		$screen = wc_get_container()->get( CustomOrdersTableController::class )->custom_orders_table_usage_is_enabled() ? wc_get_page_screen_id( 'shop-order' ) : 'shop_order';
```
Comment (DavidAnderson684) [inline includes/class-wcpdf-admin.php:319]: @alexmigf Won't there be a fatal error for the non-existent class `CustomOrdersTableController` before WC 7.1?

**Review by DavidAnderson684 — COMMENTED** [review]
(I've only read the code, not tried to run it).

File: assets/css/order-styles.css [hunk 3]
```diff
@@ -21,8 +23,9 @@
 	vertical-align: middle;
 }
 
-.type-shop_order .column-wc_actions a.button.wpo_wcpdf img,
-.type-shop_order .column-order_actions a.button.wpo_wcpdf img /* legacy: 3.2 & older */ {
+.type-shop_order .column-order_actions a.button.wpo_wcpdf img, /* legacy: 3.2 & older */
+.type-shop_order .column-wc_actions a.button.wpo_wcpdf img, /* pre WC 7.1 */
+.woocommerce_page_wc-orders .column-wc_actions a.button.wpo_wcpdf img {
 	width: 16px;
 }
```
Comment (Terminator-Barbapapa) [inline assets/css/order-styles.css:30]: There is still a difference between the buttons pre-HPOS and after. The icons are not correctly centered.

Before:
Schermafbeelding 2022-12-02 om 16 45 32

After:
Schermafbeelding 2022-12-02 om 16 45 24

**Review by Terminator-Barbapapa — CHANGES_REQUESTED** [review]

File: assets/css/order-styles.css [hunk 4]
```diff
@@ -1,29 +1,26 @@
...
-
-.wpo_wcpdf-actions a.button.exists::after {
-	font-family: Dashicons;
-	content: "\f147";
-	font-size: 125%;
-	line-height: 80%;
-	margin-left: 4px;
-	color: #2aad2a;
-	vertical-align: middle;
-}
```
[the remark this answers is not in the document]
Comment (alexmigf) [reply assets/css/order-styles.css:22]: Ouch

**Comment by passatgt** [comment]
Bulk actions handling is different with HPOS too: https://github.com/woocommerce/woocommerce/pull/35442

**Comment by alexmigf** [comment]
@passatgt thank you, I will take a look.

**Review by Terminator-Barbapapa — APPROVED** [review]

**Comment by dpeyou** [comment]
So far, the only error that I encounter is with HPOS enabled + Sync enabled. In this instance:
- A certain order never appears as generated
- Each time I refresh the page when viewing this invoice, the Invoice number increments by 1

Invoice has already been generated, but the invoice number doesn't appear & the invoice icon doesn't have the green tick


On page refresh

**Comment by dpeyou** [comment]
I don't know how relevant this is, but for order #113 that I mentioned in my previous comment, when I disabled HPOS, I noticed that only for this particular order, the Invoice number disappeared (from Orders screen) and the green tick on the invoice number was gone. This didn't happen with any other order. When I generated the invoice, it incremented by 1: In my previous comment I stopped at 14. This time, the newly generated number is 15. When I refresh the page however, it does not increment the Invoice number :+1:

**Comment by alexmigf** [comment]
Sync is not ready yet, almost sure that document meta wasn't saved in that order. If you didn't encounter other issues without sync it should be ok.

**Review by dpeyou — APPROVED** [review]

**Comment by dwalkerpriv** [comment]
It won't work with my installation PDF invoices. Is there a particular version I should be using? I've updated to the latest version

Screenshot 2023-01-10 093539
Screenshot 2023-01-10 093550

**Comment by dwalkerpriv** [comment]
i haven't had any issues printing the data with:
- sync option on
- sync option off
- HPOS on
- HPOS off

Screenshot 2023-01-11 092717

Screenshot 2023-01-11 093433

**Review by dwalkerpriv — APPROVED** [review]

**Comment by jhosagid7** [comment]
# HPOS compatibility

I also had no problems printing the data

Just to mention that at the beginning I had installed the plugins "PDF Invoices", "Stripe" and it showed me this warning:
Advertiencia-plugin-existentes

After I disabled those plugins, it did show me the option for "High Performance Order Storage":
habilitada la opcion despues de desactivar los plugins


- Existing WC <=7.0 setup + WC 7.1 upgrade + HPOS enabled + Sync enabled (can have issues) ✔️ 
- New WC 7.1 setup + HPOS enabled + Sync disabled ✔️ 
- Existing WC <7.1 setup + HPOS disabled ✔️ 
- New WC 7.1 setup + HPOS disabled ✔️ 

invoice

Packing slips

**Review by jhosagid7 — APPROVED** [review]

**Review by YordanSoares — APPROVED** [review]
Tests performed:
1. Installed WooCommerce and created 3 orders using the WordPress posts table;
2. Created PDF invoices for the 3 orders created in the previous step;
3. Enabling HPOS, created new High-Performance Order tables, and selected the WooCommerce order tables as default data store;
4. Created 3 new orders and PDF invoices for these orders;
5. All the orders and PDF invoices opens correctly.

The only different thing I realized is the icon update: 

image

**Comment by alexmigf** [comment]
@YordanSoares did you test the bulk action?

**Comment by YordanSoares** [comment]
@alexmigf yes, it works smoothly, to open existing and generate new documents.

**Pull request merged**